openflow15/openflow15.go

```diff
@@ -0,0 +1,2544 @@
+	Type_GetConfigRequest = 7
+	Type_GetConfigReply   = 8
+	Type_SetConfig        = 9
+
+	/* Asynchronous messages. */
+	Type_PacketIn    = 10
+	Type_FlowRemoved = 11
+	Type_PortStatus  = 12
+
+	/* Controller command messages. */
+	Type_PacketOut = 13
+	Type_FlowMod   = 14
+	Type_GroupMod  = 15
+	Type_PortMod   = 16
+	Type_TableMod  = 17
+
+	/* Multipart messages. */
+	Type_MultiPartRequest = 18
+	Type_MultiPartReply   = 19
+
+	/* Barrier messages. */
+	Type_BarrierRequest = 20
+	Type_BarrierReply   = 21
+
+	/* Queue Configuration messages. */
+	Type_QueueGetConfigRequest = 22
+	Type_QueueGetConfigReply   = 23
+
+	/* Controller role change request messages. */
+	Type_RoleRequest = 24
+	Type_RoleReply   = 25
+
+	/* Asynchronous message configuration. */
+	Type_GetAsyncRequest = 26
+	Type_GetAsyncReply   = 27
+	Type_SetAsync        = 28
+
+	/* Meters and rate limiters configuration messages. */
+	Type_MeterMod = 29
+
+    /* Controller role change event messages. */
+    Type_RoleStatus = 30
+
+    /* Asynchronous messages. */
+    Type_TableStatus = 31
+
+    /* Request forwarding by the switch. */
+    Type_RequestForward = 32
+
+    /* Bundle operations (multiple messages as a single operation). */
+    Type_BundleControl    = 33
+    Type_BundleAddMessage = 34
+
+    /* Controller Status async message. */
+    Type_ControllerStatus = 35
+)
+
+func Parse(b []byte) (message util.Message, err error) {
+	fmt.Println("Parse_Type=", b[1])
+	switch b[1] {
+	case Type_Hello:
+		message = new(common.Hello)
+		err = message.UnmarshalBinary(b)
+	case Type_Error:
+		errMsg := new(ErrorMsg)
+		err = errMsg.UnmarshalBinary(b)
+		if err != nil {
+			return
+		}
+		switch errMsg.Type {
+		case ET_EXPERIMENTER:
+			message = new(VendorError)
+			err = message.UnmarshalBinary(b)
+		default:
+			message = errMsg
+		}
+	case Type_EchoRequest:
+		message = new(common.Header)
+		err = message.UnmarshalBinary(b)
+	case Type_EchoReply:
+		message = new(common.Header)
+		err = message.UnmarshalBinary(b)
+	case Type_Experimenter:
+		message = new(VendorHeader)
+		err = message.UnmarshalBinary(b)
+	case Type_FeaturesRequest:
+		message = NewFeaturesRequest()
+		err = message.UnmarshalBinary(b)
+	case Type_FeaturesReply:
+		message = NewFeaturesReply()
+		err = message.UnmarshalBinary(b)
+	case Type_GetConfigRequest:
+		message = new(common.Header)
+		err = message.UnmarshalBinary(b)
+	case Type_GetConfigReply:
+		message = new(SwitchConfig)
+		err = message.UnmarshalBinary(b)
+	case Type_SetConfig:
+		message = NewSetConfig()
+		err = message.UnmarshalBinary(b)
+	case Type_PacketIn:
+		message = new(PacketIn)
+		err = message.UnmarshalBinary(b)
+	case Type_FlowRemoved:
+		message = NewFlowRemoved()
+		err = message.UnmarshalBinary(b)
+	case Type_PortStatus:
+		message = NewPortStatus()
+		err = message.UnmarshalBinary(b)
+	case Type_PacketOut:
+		break
+	case Type_FlowMod:
+		message = NewFlowMod()
+		err = message.UnmarshalBinary(b)
+	case Type_GroupMod:
+		break
```

Comment:
Just curious GroupMod/PortMod/TableMod is not supported in OF1.5?

Reply:
They are. Forgot to add it in the Parse() function. Would add.